rust/src/http2/http2.rs

```diff
@@ -0,0 +1,1175 @@
+                        );
+                        break;
+                    }
+                }
+            }
+            None => panic!("BUG"),
+        }
+    }
+
+    fn parse_ts(&mut self, mut input: &[u8]) -> AppLayerResult {
+        //first consume frame bytes
+        let il = input.len();
+        if self.request_frame_size > 0 {
+            let ilen = input.len() as u32;
+            if self.request_frame_size >= ilen {
+                self.request_frame_size -= ilen;
+                return AppLayerResult::ok();
+            } else {
+                let start = self.request_frame_size as usize;
+                input = &input[start..];
+                self.request_frame_size = 0;
+            }
+        }
+
+        //then parse all we can
+        while input.len() > 0 {
+            match parser::http2_parse_frame_header(input) {
+                Ok((rem, head)) => {
+                    let hl = head.length as usize;
+
+                    //we check for completeness first
+                    if rem.len() < hl {
+                        //but limit ourselves so as not to exhaust memory
+                        if hl < HTTP2_MAX_HANDLED_FRAME_SIZE {
+                            return AppLayerResult::incomplete(
+                                (il - input.len()) as u32,
+                                (HTTP2_FRAME_HEADER_LEN + hl) as u32,
+                            );
+                        } else if rem.len() < HTTP2_MIN_HANDLED_FRAME_SIZE {
+                            return AppLayerResult::incomplete(
+                                (il - input.len()) as u32,
+                                (HTTP2_FRAME_HEADER_LEN + HTTP2_MIN_HANDLED_FRAME_SIZE) as u32,
+                            );
+                        } else {
+                            self.set_event(HTTP2Event::LongFrameData);
+                            self.request_frame_size = head.length - (rem.len() as u32);
+                        }
+                    }
+
+                    //get a safe length for the buffer
+                    let (hlsafe, complete) = if rem.len() < hl {
+                        (rem.len(), false)
+                    } else {
+                        (hl, true)
+                    };
+
+                    if head.length == 0 && head.ftype == parser::HTTP2FrameType::SETTINGS as u8 {
+                        input = &rem[hlsafe..];
+                        continue;
+                    }
+                    let txdata = self.parse_frame_data(
+                        head.ftype,
+                        &rem[..hlsafe],
+                        complete,
+                        head.flags,
+                        STREAM_TOSERVER,
+                    );
+
+                    let tx = self.find_or_create_tx(&head, &txdata, STREAM_TOSERVER);
+                    tx.handle_frame(&head, &txdata, STREAM_TOSERVER);
+                    let over = head.flags & parser::HTTP2_FLAG_HEADER_END_STREAM != 0;
```

Comment:
maybe name this something like `eof` (or `eos` ;) )

Reply:
of for `EOS`